src/rng.cpp

```diff
@@ -5,35 +5,49 @@
 #include <algorithm>
 #include <utility>
 
-long rng( long val1, long val2 )
+#include "cata_utility.h"
+
+static std::uniform_int_distribution<int> rng_int_dist;
+static std::uniform_real_distribution<double> rng_real_dist;
+static std::normal_distribution<double> rng_normal_dist;
+static std::uniform_int_distribution<unsigned int>
+rng_uint_dist( 0, std::numeric_limits<unsigned int>::max() );
+
+unsigned int rng_bits()
 {
-    long minVal = ( val1 < val2 ) ? val1 : val2;
-    long maxVal = ( val1 < val2 ) ? val2 : val1;
-    return minVal + long( ( maxVal - minVal + 1 ) * double( rand() / double( RAND_MAX + 1.0 ) ) );
+    return rng_uint_dist( rng_get_engine() );
 }
 
-double rng_float( double val1, double val2 )
+int rng( int lo, int hi )
 {
-    double minVal = ( val1 < val2 ) ? val1 : val2;
-    double maxVal = ( val1 < val2 ) ? val2 : val1;
-    return minVal + ( maxVal - minVal ) * static_cast<double>( rand() ) / static_cast<double>
-           ( RAND_MAX + 1.0 );
+    if( lo > hi ) {
+        std::swap( lo, hi );
+    }
+
+    decltype( rng_int_dist.param() ) range( lo, hi );
+    rng_int_dist.param( range );
```

Comment:
Now you're modifying the global variable I'm even more uncomfortable.  I'd say just make a local distribution object.  Constructing a new one should be basically the same cost as re-parameterising an existing one.

Reply:
There's an option to do a per-invocation override, so I did that.